content/docs/1.13/performance-tuning.md

```diff
@@ -0,0 +1,87 @@
+
+### Workers
+
+Items from the span queue in the collector are picked up by workers. Each worker picks one span from the queue and persists it to the storage. The number of workers can be specified by the setting `collector.num-workers`(default: `50`) and should be as high as needed to keep the queue close to zero. The general rule is: the faster the backing storage, the lower the number of workers can be. Given that workers are relatively cheap, this number can be increased at will. As a general rule, one worker per 50 items in the queue should be sufficient when the storage is fast. With a `collector.queue-size` of `2000`, having about `40` workers should be sufficient. For slower storage mechanisms, this ratio should be adjusted accordingly, having more workers per queue item.
+
+## Deployment architecture
+
+Although performance tuning the individual components is important, the way Jaeger is deployed can be decisive in obtaining optimal performance.
+
+### Scale the collector up and down
+
+Use the auto-scaling capabilities of your platform: the collector is nearly horizontally scalable so that more instances can be added and removed on-demand. A good way to scale up and down is by checking the `jaeger_collector_queue_length` metric: add instances when the length is higher than 50% of the maximum size for extended periods of time. Another metric that can be taken into consideration is `jaeger_collector_in_queue_latency_bucket`, which is a histogram indicating how long spans have been waiting in the queue before a worker picked it up. When the queue latency gets higher over time, it’s a good indication to increase the number of the workers, or to improve the storage performance.
+
+### Make sure the storage can keep up
+
+Each span is written to the storage by the collector using one worker, blocking it until the span has been stored. When the storage is too slow, the number of workers blocked by the storage might be too high, causing spans to be dropped. To help diagnose this situation, the histogram `jaeger_collector_save_latency_bucket` can be analyzed. Ideally, this histogram would show a nearly constant value. When the histogram shows that most spans are taking longer and longer over time, it’s a good indication that your storage might need some attention.
+
+### Place the agents close to your applications
+
+The agent is meant to be placed on the same host as the instrumented application. This is typically accomplished by having one agent per bare metal for traditional applications, or as a sidecar in container environments like Kubernetes, as this helps spread the load handled by agents with the additional advantage of allowing each agent to be tweaked individually, according to the application’s needs and importance.
+
+### Consider using Apache Kafka as the temporary storage
```

Comment:
2nd level header, not 3rd. People complained about calling Kafka "storage", so "intermediate buffer".

Reply:
It's third-level because it's inside "Deployment architecture".